api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -98,6 +98,15 @@
    */
   ExpireSnapshots executeDeleteWith(ExecutorService executorService);
 
+  /**
+   * Passes an alternative executor service that will be used for accessing manifests.
+   * If this method is not called, the default worker pool will be used.
+   *
+   * @param executorService an executor service to access manifests
+   * @return this for method chaining
+   */
+  ExpireSnapshots accessManifestsWith(ExecutorService executorService);
```

Comment:
The methods that run this are named `plan`: `planFiles` and `planSplits` so I'd probably change this to `planWith`. Is that okay with you, @yittg?